core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -339,4 +347,8 @@ public boolean matchesIndices(String... indices) {
         }
         return false;
     }
+
+    public TermsLookupFetchService termsLookupFetchService() {
```

Comment:
can we have the handleTermsLookup method directly here so we don't have to spread the TermsLookupFetchService dependency around. No need or a public getter then